libmscore/edit.cpp

```diff
@@ -4302,6 +4302,9 @@ void Score::undoRemoveElement(Element* element)
                   if (!segments.contains(s))
                         segments.append(s);
                   }
+            if (e->parent() && e->parent()->isSystem()) {
+                  e->setParent(0); // systems will be regnerated upon redo, so detach
```

Comment:
I need to know if this is proper thing to do.

Reply:
I don't know but the comment has a typo...